src/ol/proj/proj.js

```diff
@@ -675,14 +675,19 @@ ol.proj.get = function(projectionLike) {
  * @return {boolean} Equivalent.
  */
 ol.proj.equivalent = function(projection1, projection2) {
+  var units1 = projection1.getUnits();
+  var units2 = projection2.getUnits();
+  var tilePixels = ol.proj.Units.TILE_PIXELS;
+  var equalUnits = units1 === units2 ||
+      units1 === tilePixels || units2 === tilePixels;
```

Comment:
For me this is a special case that rather should be handled in `ol.featureloader.tile`. I find it kind of counter intuitive that two projections should be equivalent when only one of them has pixel units.